h2o-core/src/main/java/hex/ModelMojoWriter.java

```diff
@@ -0,0 +1,225 @@
+
+  /**
+   * Write a simple value to the model.ini/[info] section. Here "simple" means a value that can be stringified with
+   * .toString(), and its stringified version does not span multiple lines.
+   */
+  protected final void writekv(String key, Object value) throws IOException {
+    String valStr = value == null? "null" : value.toString();
+    if (valStr.contains("\n"))
+      throw new IOException("The `value` must not contain newline characters, got: " + valStr);
+    if (lkv.containsKey(key))
+      throw new IOException("Key " + key + " was already written");
+    lkv.put(key, valStr);
+  }
+
+  /** Write a binary file to the MOJO archive. */
+  protected final void writeblob(String filename, byte[] blob) throws IOException {
+    ZipEntry archiveEntry = new ZipEntry(filename);
+    archiveEntry.setSize(blob.length);
+    zos.putNextEntry(archiveEntry);
+    zos.write(blob);
+    zos.closeEntry();
+  }
+
+  /** Write a text file to the MOJO archive (or rather open such file for writing). */
+  protected final void startWritingTextFile(String filename) {
+    assert tmpfile == null : "Previous text file was not closed";
+    tmpfile = new StringBuilder();
+    tmpname = filename;
+  }
+
+  /** Write a single line of text to a previously opened text file. */
+  protected final void writeln(String s) {
+    assert tmpfile != null : "No text file is currently being written";
+    tmpfile.append(s);
+    tmpfile.append('\n');
+  }
+
+  /** Finish writing a text file. */
+  protected final void finishWritingTextFile() throws IOException {
+    assert tmpfile != null : "No text file is currently being written";
+    writeblob(tmpname, tmpfile.toString().getBytes(Charset.forName("UTF-8")));
+    tmpfile = null;
+  }
+
+
+  //--------------------------------------------------------------------------------------------------------------------
+  // Private
+  //--------------------------------------------------------------------------------------------------------------------
+
+  /**
+   * Used from `ModelsHandler.fetchMojo()` to serialize the Mojo into a StreamingSchema.
+   * The structure of the zip will be the following:
+   *    model.ini
+   *    domains/
+   *        d000.txt
+   *        d001.txt
+   *        ...
+   *    (extra model files written by the subclasses)
+   * Each domain file is a plain text file with one line per category (not quoted).
+   */
+  @Override public final void writeTo(OutputStream os) {
+    zos = new ZipOutputStream(os);
+    try {
+      addCommonModelInfo();
+      writeModelData();
+      writeModelInfo();
+      writeDomains();
+      zos.close();
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private void addCommonModelInfo() throws IOException {
+    int n_categoricals = 0;
+    for (String[] domain : model._output._domains)
+      if (domain != null)
+        n_categoricals++;
+
+    writekv("h2o_version", H2O.ABV.projectVersion());
```

Comment:
Do we need a customer ini file format? Can we instead use standard java properties files (and sparse ourselves the trouble of writing/parsing this format)?